### PR TITLE
Drag Update

### PR DESCRIPTION
**Pull request by indy91**
This update adds airfoil defintions for the CSM, S-IVB and CSM+S-IVB. All these vehicles have realistic drag functions. The CSM also simulation aerodynamic moment coefficients.

The RTCC can't fully take into account realistic drag yet, so the total drag coefficient is only 5% of what would be realistic. That is equivalent to the fixed drag coefficient of 0.1 (instead of 2.0) currently used in the RTCC. Once the RTCC is better prepared to simulate the drag, too, the change will be made to the full 100% of realistic drag. The 5% value roughly agrees with the amount the vehicles get on average right now.

### CODE REVIEW

File: Orbitersdk/samples/ProjectApollo/src_csm/saturnmesh.cpp [hunk 1]
```diff
@@ -271,6 +271,105 @@ void CMLETHoriCoeffFunc(double aoa, double M, double Re, double *cl, double *cm,
...
+	int i, j;
+	const int nabsc = 9;
+	static const double AOA[nabsc] = { 0 * RAD, 10 * RAD, 20 * RAD, 30 * RAD, 90 * RAD, 150 * RAD, 160 * RAD, 170 * RAD, 180 * RAD };
+	static const double CD_free[nabsc] = { 2.41, 2.73, 3.14, 3.61, 5.01, 2.92, 2.46, 2.08, 1.96}; //free flow
+	static const double CD_cont[nabsc] = { 0.54, 0.59, 0.77, 1.12, 2.78, 1.32, 1.41, 1.50, 1.69 }; //continuum flow
+	static const double CM_free[nabsc] = { 0, 0.0440, -0.1436, -0.4349, -0.7682, -0.2369, 0.3820, -0.0115, 0 };
+	static const double CM_cont[nabsc] = { 0, 0.0977, 0.1069, 0.0230, -0.5803, 0.0206, 0.0981, 0.0145, 0 };
+
+	//For drag
+	for (i = 0; i < nabsc - 1 && AOA[i + 1] < aoa_T; i++);
```
Comment (ThymoNL) [inline Orbitersdk/samples/ProjectApollo/src_csm/saturnmesh.cpp]: It took me a while to understand what you are doing here. While a loop without body is totally valid code, it does increase the cognitive load of the person reading it as it is not obvious that you use i later on.

To make it clear you are using this value how about doing it in the loop body? Something like:
```c
int someVar;
for (int i = 0; i < x ; i++)
    someVar++;
```
The compiler will most likely optimize it away (if you are worried about that) nor will it affect complexity.
Alternatively, I would at the very least comment what you are doing and change i to a descriptive name.

File: Orbitersdk/samples/ProjectApollo/src_csm/saturnmesh.cpp [hunk 2]
```diff
@@ -271,6 +271,105 @@ void CMLETHoriCoeffFunc(double aoa, double M, double Re, double *cl, double *cm,
...
+	double f = (aoa_T - AOA[i]) / (AOA[i + 1] - AOA[i]);
+
+	//For moments
+	for (j = 0; j < nabsc - 1 && AOA[j + 1] < abs(aoa); j++);
```
Comment (ThymoNL) [inline Orbitersdk/samples/ProjectApollo/src_csm/saturnmesh.cpp]: Same here

File: Orbitersdk/samples/ProjectApollo/src_csm/saturnmesh.cpp [hunk 3]
```diff
@@ -271,6 +271,105 @@ void CMLETHoriCoeffFunc(double aoa, double M, double Re, double *cl, double *cm,
...
+	*cd = 0;
+
+	const int nabsc = 9;
+	static const double AOA[nabsc] = { 0 * RAD, 10 * RAD, 20 * RAD, 30 * RAD, 90 * RAD, 150 * RAD, 160 * RAD, 170 * RAD, 180 * RAD };
+	static const double CM_free[nabsc] = { 0, -0.0440, 0.1436, 0.4349, 0.7682, 0.2369, -0.3820, 0.0115, 0 };
+	static const double CM_cont[nabsc] = { 0, -0.0977, -0.1069, -0.0230, 0.5803, -0.0206, -0.0981, -0.0145, 0 };
+
+	double Kn = M / Re * 1.482941286; //Knudsen number. Factor is sqrt(1.4*pi/2)
+	int i;
+	for (i = 0; i < nabsc - 1 && AOA[i + 1] < abs(aoa); i++);
```
Comment (ThymoNL) [inline Orbitersdk/samples/ProjectApollo/src_csm/saturnmesh.cpp]: Same here

File: Orbitersdk/samples/ProjectApollo/src_saturn/sat1bmesh.cpp [hunk 4]
```diff
@@ -183,6 +183,46 @@ static PARTICLESTREAMSPEC prelaunchvent_spec = {
...
+	aoa = acos(unit(vec).z);
+
+	double Kn = M / Re * 1.482941286; //Knudsen number. Factor is sqrt(1.4*pi/2)
+	int i;
+	const int nabsc = 6;
+	static const double AOA[nabsc] = { 0 * RAD, 10 * RAD,30 * RAD,90 * RAD,150 * RAD, 180 * RAD };
+	static const double CD_free[nabsc] = { 2.9251, 3.3497, 6.1147, 11.08, 6.1684, 3.1275 }; //free flow
+	static const double CD_cont[nabsc] = { 0.44, 0.59, 1.12, 2.78, 1.8, 1.5 }; //continuum flow
+
+	for (i = 0; i < nabsc - 1 && AOA[i + 1] < aoa; i++);
```
Comment (ThymoNL) [inline Orbitersdk/samples/ProjectApollo/src_saturn/sat1bmesh.cpp]: Same here

File: Orbitersdk/samples/ProjectApollo/src_saturn/sat5mesh.cpp [hunk 5]
```diff
@@ -52,6 +52,44 @@
...
+	double Kn = M / Re * 1.482941286; //Knudsen number. Factor is sqrt(1.4*pi/2)
+	int i;
+	const int nabsc = 9;
+	static const double AOA[nabsc] = { -180 * RAD, -90 * RAD,-30 * RAD, -10 * RAD,0 * RAD, 10 * RAD,30 * RAD,90 * RAD,180 * RAD };
+	//static const double CL[nabsc] = { 0,      0,   -0.004,     0,     0.008,     0,      0 };
+	static const double CD_free[nabsc] = { 3.1275, 11.08, 6.1147, 3.3497, 2.9251, 3.3497, 6.1147, 11.08, 3.1275 }; //free flow
+	static const double CD_cont[nabsc] = { 1.69, 2.78, 1.12, 0.59, 0.54, 0.59, 1.12, 2.78, 1.69 }; //continuum flow
+	//static const double CM[nabsc] = { 0,      0,   0.0014,  0,-0.0012,     0,      0 };
+
+	for (i = 0; i < nabsc - 1 && AOA[i + 1] < aoa; i++);
```
Comment (ThymoNL) [inline Orbitersdk/samples/ProjectApollo/src_saturn/sat5mesh.cpp]: Same here

File: Orbitersdk/samples/ProjectApollo/src_saturn/sivb.cpp [hunk 6]
```diff
@@ -153,6 +153,47 @@ void SIVbLoadMeshes()
...
+	aoa = acos(unit(vec).z);
+
+	double Kn = M / Re * 1.482941286; //Knudsen number. Factor is sqrt(1.4*pi/2)
+	int i;
+	const int nabsc = 6;
+	static const double AOA[nabsc] = { 0 * RAD, 10 * RAD,30 * RAD,90 * RAD,150 * RAD, 180 * RAD };
+	static const double CD_free[nabsc] = { 2.9251, 3.3497, 6.1147, 11.08, 6.1684, 3.1275 }; //free flow
+	static const double CD_cont[nabsc] = { 0.44, 0.59, 1.12, 2.78, 1.8, 1.5 }; //continuum flow
+
+	for (i = 0; i < nabsc - 1 && AOA[i + 1] < aoa; i++);
```
Comment (ThymoNL) [inline Orbitersdk/samples/ProjectApollo/src_saturn/sivb.cpp]: Same here

**Review by ThymoNL — CHANGES_REQUESTED** [review]

**Review by ThymoNL — APPROVED** [review]

**Review by n7275 — APPROVED** [review]

**Pull request merged**